protocol/decoupled-hardfork-features.md

```diff
@@ -0,0 +1,308 @@
+# Purpose
+
+The purpose of this design document is to propose a mechanism for decoupling
+individual features from their hardforks in the OP Stack. This will enhance
+flexibility in feature activation, enabling granular control for testing,
+custom chain configurations, and phased rollouts while maintaining
+coordination across the superchain. The design aims to improve modularity
```

Comment:
> custom chain configurations

Is there a concern that some chains may activate features in different forks, leading to pathological configurations that are untested and unaudited? i.e., even if a set of features is sound and reasonable, a subset may not be.

I'm not sure how this interacts with the definition of "standard config", but at first glance it seems prudent to mandate that the same features be deployed at the same forks.